백다연/MvvmStudy/app/src/main/java/com/test/mvvmstudy/view/SearchFragment.kt

```diff
@@ -0,0 +1,88 @@
+package com.test.mvvmstudy.view
+
+import android.os.Bundle
+import android.view.*
+import androidx.appcompat.widget.SearchView
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.Navigation
+import com.test.mvvmstudy.R
+import com.test.mvvmstudy.adapter.SearchResultAdapter
+import com.test.mvvmstudy.databinding.FragmentSearchBinding
+import com.test.mvvmstudy.data.ResultDetail
+import com.test.mvvmstudy.viewmodel.SearchViewModel
+
+class SearchFragment : Fragment() {
+    private lateinit var binding: FragmentSearchBinding
+    private val adapter: SearchResultAdapter by lazy {
+        SearchResultAdapter()
+    }
+    private val searchViewModel: SearchViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentSearchBinding.inflate(inflater, container, false)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setHasOptionsMenu(true)
+        setRecyclerviewAdapter()
+        adapterClickListener()
+        initObserver()
+    }
+
+    private fun setRecyclerviewAdapter() {
+        binding.searchRecyclerview.adapter = adapter
+    }
+
+    private fun adapterClickListener() {
+        adapter.itemClick = object : SearchResultAdapter.ItemClick {
+            override fun onClick(view: View, searchData: ResultDetail) {
+                val bundle = Bundle()
+                bundle.putSerializable("searchData", searchData)
+                Navigation.findNavController(view)
+                    .navigate(R.id.action_searchFragment_to_searchDetailFragment, bundle)
+            }
+        }
+    }
+
+    private fun initObserver() {
+        searchViewModel.isLoading.observe(viewLifecycleOwner) { status ->
+            if (status) binding.progressBar.visibility = View.VISIBLE
+            else binding.progressBar.visibility = View.INVISIBLE
```

Comment:
여기도 한줄로 줄일 수 있을 것 같아요!
```suggestion
            binding.progressBar.isVisible = status
```

Reply:
넵! 합치는게 깔끔할 것 같네요!!